src/Test/Perf/util.csx

```diff
@@ -0,0 +1,180 @@
+//
+
+/// Returns the directory that houses the currenly executing script.
+public string MyWorkingDirectory([CallerFilePath] string sourceFilePath = "")
+{
+    return Directory.GetParent(sourceFilePath).FullName;
+}
+
+public string MyArtifactsDirectory([CallerFilePath] string sourceFilePath = "")
+{
+    var path = Path.Combine(MyWorkingDirectory(sourceFilePath), "artifacts");
+    Directory.CreateDirectory(path);
+    return path;
+}
+
+public string MyResultsDirectory([CallerFilePath] string sourceFilePath = "")
+{
+    var path = Path.Combine(MyWorkingDirectory(sourceFilePath), "results");
+    Directory.CreateDirectory(path);
+    return path;
+}
+
+public string RoslynDirectory([CallerFilePath] string sourceFilePath = "")
+{
+    var workingDir = MyWorkingDirectory(sourceFilePath);
+    var srcTestPerf = Path.Combine("src", "Test", "Perf").ToString();
+    return workingDir.Substring(0, workingDir.IndexOf(srcTestPerf));
+}
+
+public string BinDebugDirectory([CallerFilePath] string sourceFilePath = "")
+{
+    return Path.Combine(RoslynDirectory(), "Binaries", "Debug");
+}
+
+public string BinReleaseDirectory([CallerFilePath] string sourceFilePath = "")
+{
+    return Path.Combine(RoslynDirectory(), "Binaries", "Release");
+}
+
+
+//
+// C# scripting utility 
+//
+
+/// Finds all csi files in a directory recursively, ignoring those that
+/// are in the "skip" set.
+public IEnumerable<string> allCsiRecursive(string start, HashSet<string> skip)
+{
+    IEnumerable<string> childFiles =
+        from fileName in Directory.EnumerateFiles(start.ToString(), "*.csx")
+        select fileName;
+    IEnumerable<string> grandChildren =
+        from childDir in Directory.EnumerateDirectories(start)
+        where !skip.Contains(childDir)
+        from fileName in allCsiRecursive(childDir, skip)
+        select fileName;
+    return childFiles.Concat(grandChildren).Where((s) => !skip.Contains(s));
+}
+
+/// Runs the script at fileName and returns a task containing the
+/// state of the script.
+public async Task<ScriptState<int>> RunFile(string fileName)
+{
+    var scriptOptions = ScriptOptions.Default.WithFilePath(fileName);
+    var text = File.ReadAllText(fileName);
+    var prelude = "public System.Collections.Generic.List<string> Args = null;";
+    var state = await CSharpScript.RunAsync(prelude);
+    var args = state.GetVariable("Args");
+    args.Value = Args;
+    return await state.ContinueWithAsync<int>(text, scriptOptions);
+}
+
+//
+// Process spawning and error handling.
+//
+
+public class ProcessResult
+{
+    public string ExecutablePath {get; set;}
+    public string Args {get; set;}
+    public int Code {get; set;}
+    public string StdOut {get; set;}
+    public string StdErr {get; set;}
+
+    public bool Failed  => Code != 0;
+    public bool Succeeded  => !Failed;
+}
+
+public ProcessResult ShellOut(
+        string file,
+        string args,
+        CancellationToken? cancelationToken = null)
```

Comment:
What I mean is how do we envision plumbing a CancellationToken down to this point, and who will be calling CancellationTokenSource.Cancel?  My initial thought is that we would want this to happen when someone hits Ctrl+C in the console.  How will we wire this up (or is this intended for some other purpose)?